### PR TITLE
Fix GCM benchmarks and add to Slurm CI

### PR DESCRIPTION
**Pull request by thomasgibson**
# Description

The newly added GCM benchmarks on master are already broken. This PR hopefully fixes any lingering issues and adds _all_ gcm tests to the Slurm CI suite (on GPU)

I have

- [x] Written and run all necessary tests with CLIMA by including `tests/runtests.jl`
- [x] Followed all necessary [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) and run `julia .dev/climaformat.jl .`
- [x] Updated the documentation to reflect changes from this PR.

# For review by CLIMA developers

- [x] There are no open pull requests for this already
- [x] CLIMA developers with relevant expertise have been assigned to review this submission
- [x] The code conforms to the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) and has consistent naming conventions. `julia .dev/format.jl` has been run in a separate commit.
- [x] This code does what it is technically intended to do (all numerics make sense physically and/or computationally)

### CODE REVIEW

**Comment by thomasgibson** [comment]
bors try

**Comment by jakebolewski** [comment]
Will these experiments write output by default? We might want to disable that, or redirect write output to `/dev/null`

**Comment by thomasgibson** [comment]
Good point --- we don't want any output; I will make sure they don't and update

**Comment by thomasgibson** [comment]
bors try

**Comment by thomasgibson** [comment]
bors try

**Comment by thomasgibson** [comment]
bors try

File: slurmci-test.toml [hunk 1]
```diff
@@ -57,10 +57,12 @@ gpu = [
   { file = "test/Numerics/DGMethods/advection_diffusion/periodic_3D_hyperdiffusion.jl", slurmargs = ["--ntasks=3"], args = ["--integration-testing"] },
   { file = "tutorials/Atmos/dry_rayleigh_benard.jl", slurmargs = ["--ntasks=1"], args = ["--fix-rng-seed"] },
   { file = "tutorials/Atmos/risingbubble.jl", slurmargs = ["--ntasks=3"], args = [] },
-  { file = "experiments/AtmosGCM/heldsuarez.jl", slurmargs = ["--ntasks=1"], args = ["--diagnostics=default", "--fix-rng-seed"] },
+  { file = "experiments/AtmosGCM/heldsuarez.jl", slurmargs = ["--ntasks=1"], args = ["--diagnostics=never", "--fix-rng-seed"] },
+  { file = "experiments/AtmosGCM/baroclinic-wave.jl", slurmargs = ["--ntasks=1"], args = ["--diagnostics=never"] },
```
Comment (kpamnany) [inline slurmci-test.toml:61]: Don't need `--diagnostics=never`, that's the default (unless you're calling `ClimateMachine.init(diagnostics = <interval>)`).

File: slurmci-test.toml [hunk 2]
```diff
@@ -58,6 +58,9 @@ gpu = [
   { file = "tutorials/Atmos/dry_rayleigh_benard.jl", slurmargs = ["--ntasks=1"], args = ["--fix-rng-seed"] },
   { file = "tutorials/Atmos/risingbubble.jl", slurmargs = ["--ntasks=3"], args = [] },
   { file = "experiments/AtmosGCM/heldsuarez.jl", slurmargs = ["--ntasks=1"], args = ["--diagnostics=default", "--fix-rng-seed"] },
+  { file = "experiments/AtmosGCM/baroclinic-wave.jl", slurmargs = ["--ntasks=1"], args = ["--diagnostics=default"] },
+  { file = "experiments/AtmosGCM/nonhydrostatic-gravity-wave.jl", slurmargs = ["--ntasks=1"], args = ["--diagnostics=default"] },
+  { file = "experiments/AtmosGCM/isothermal-zonal-flow.jl", slurmargs = ["--ntasks=1"], args = ["--diagnostics=default"] },
```
Comment (kpamnany) [inline slurmci-test.toml:63]:
```suggestion
  { file = "experiments/AtmosGCM/baroclinic-wave.jl", slurmargs = ["--ntasks=1"], args = [] },
  { file = "experiments/AtmosGCM/nonhydrostatic-gravity-wave.jl", slurmargs = ["--ntasks=1"], args = [] },
  { file = "experiments/AtmosGCM/isothermal-zonal-flow.jl", slurmargs = ["--ntasks=1"], args = [] },
```
Diagnostics are enabled (with `--diagnostics=default`) in Held-Suarez to test the GCM diagnostics (the interval was chosen such that diagnostics output would happen only once). There's no need to test the same code path more than once.

**Comment by thomasgibson** [comment]
bors try

**Pull request merged**